python/src/iceberg/transforms.py

```diff
@@ -120,9 +120,6 @@ def hash(self, value: S) -> int:
         raise NotImplementedError()
 
     def apply(self, value: S) -> Optional[int]:
-        if value is None:
```

Comment:
Since we have `S`, it can never be `None`, otherwise, the signature should be `Optional[S]`. If we want, we can still check if it `None`, `""` or `[]` using:
```python
if not value:
```

Reply:
Okay, looking at the tests, we _might_ expect a `None` there. I've changed the code the other way around.

Reply:
Yeah, transforms can be applied to optional columns so this should be `Optional[S]`.